### PR TITLE
Use ArgumentNullException.ThrowIfNull part 2

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
use ArgumentNullException.ThrowIfNull
Fix codefactor

## PR Context
Contributes to #19212

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/engine/remoting/client/ThrottlingJob.cs [hunk 1]
```diff
@@ -369,7 +369,10 @@ internal void DisableFlowControlForPendingCmdletActionsQueue()
         internal void AddChildJobWithoutBlocking(StartableJob childJob, ChildJobFlags flags, Action jobEnqueuedAction = null)
         {
             ArgumentNullException.ThrowIfNull(childJob);
-            if (childJob.JobStateInfo.State != JobState.NotStarted) throw new ArgumentException(RemotingErrorIdStrings.ThrottlingJobChildAlreadyRunning, nameof(childJob));
+            if (childJob.JobStateInfo.State != JobState.NotStarted) 
+            {
+                throw new ArgumentException(RemotingErrorIdStrings.ThrottlingJobChildAlreadyRunning, nameof(childJob));
+            }
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/client/ThrottlingJob.cs:375]:
```suggestion
            }

```

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs [hunk 2]
```diff
@@ -42,21 +42,15 @@ protected override void OnKeyDown(KeyEventArgs e)
 
         partial void OnSelectItemExecutedImplementation(ExecutedRoutedEventArgs e)
         {
-            if (e.Parameter == null)
-            {
-                throw new ArgumentException("e.Parameter is null", "e");
-            }
+            ArgumentNullException.ThrowIfNull(e.Parameter);
```
Comment (daxian-dbw) [inline src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs:45]: This is not a correct change, the parameter name used by `ArgumentNullException.ThrowIfNull(e.Parameter)` will be `e.Paraemter`, which is wrong.

File: src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs [hunk 3]
```diff
@@ -42,21 +42,15 @@ protected override void OnKeyDown(KeyEventArgs e)
...
             this.picker.IsOpen = false;
         }
 
         partial void OnDeleteItemExecutedImplementation(ExecutedRoutedEventArgs e)
         {
-            if (e.Parameter == null)
-            {
-                throw new ArgumentException("e.Parameter is null", "e");
-            }
+            ArgumentNullException.ThrowIfNull(e.Parameter);
```
Comment (daxian-dbw) [inline src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs:53]: same here.

File: src/Microsoft.Management.UI.Internal/ManagementList/FilterCore/FilterRules/PropertyValueSelectorFilterRule.cs [hunk 4]
```diff
@@ -76,10 +76,7 @@ public PropertyValueSelectorFilterRule(string propertyName, string propertyDispl
 
             foreach (FilterRule rule in rules)
             {
-                if (rule == null)
-                {
-                    throw new ArgumentException("A value within rules is null", "rules");
-                }
+                ArgumentNullException.ThrowIfNull(rule);
```
Comment (daxian-dbw) [inline src/Microsoft.Management.UI.Internal/ManagementList/FilterCore/FilterRules/PropertyValueSelectorFilterRule.cs:79]: Here as well. Parameter name is wrong.

**Review by daxian-dbw — COMMENTED** [review]
Some of the changes in this PR are not correct. The parameter name used for the new exception will be wrong.

File: src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs (hunk 2 above)
Comment (iSazonov) [reply src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs:45]: Example
https://source.dot.net/#Microsoft.AspNetCore.Http.Extensions/ProblemDetailsService.cs,29

Comment (daxian-dbw) [reply src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs:45]: Is that the only place you find using this API in this way?

Looking through the usage here: https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/ArgumentNullException.cs,432088dc6d98b845,references, I don't think we can call that an adopted pattern in .NET code.

Comment (daxian-dbw) [reply src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs:45]: The original `ArgumentException` is more suitable than using `ArgumentNullException.ThrowIfNull(e.Parameter)` in this case, though it's not using a resource string for its message.

**Comment by daxian-dbw** [comment]
@iSazonov and @CarloToso I will have to revert this change because some of the exception changes are incorrect -- the original exception has better error message and the right parameter name. Examples:
- https://github.com/PowerShell/PowerShell/pull/19258#discussion_r1123510784
- https://github.com/PowerShell/PowerShell/pull/19258#discussion_r1123515823
- https://github.com/PowerShell/PowerShell/pull/19258#discussion_r1123510435

@CarloToso The part of changes that fixed CodeFactor issues looks great. If you can submit that part again, I will get it merged. Thanks!

File: src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs (hunk 2 above)
Comment (iSazonov) [reply src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs:45]: Yes, you're right. For some reason I was sure they had changed the message template.
But it seems the original message is worst :-) 
```powershell
[argumentnullexception]::new("e.Parameter is null", "e")

Message        : e (Parameter 'e.Parameter is null')
ParamName      : e.Parameter is null
```

Comment (daxian-dbw) [reply src/Microsoft.Management.UI.Internal/ManagementList/Common/ListOrganizer.cs:45]: The original one is using `ArgumentException`, not `argumentnullexception`.

This is how it looks:
image

**Pull request merged**